nce/NCE-Pairwise-SM/trec_dataset.py

```diff
@@ -0,0 +1,31 @@
+from torchtext import data
+import os
+
+class TrecDataset(data.TabularDataset):
+    dirname = 'data'
+    @classmethod
+
+    def splits(cls, question_id, question_field, answer_id, answer_field, external_field, label_field,
+               train='train.tsv', validation='dev.tsv', test='test.tsv'):
+        path = './data'
+        prefix_name = 'trecqa.'
+        return super(TrecDataset, cls).splits(
+            os.path.join(path, prefix_name), train, validation, test,
```

Comment:
`os.path.join(path, prefix_name), train, validation, test` will return `./data/trecqa./train.tsv`

Reply:
@HTAustin that's because you're using the new torchtext. @Victor0118 will have to update his code to suit the new torchtext. Also, @Victor0118 update your pytorch to the recent version and enable `torch.backends.cudnn.deterministic = True`

Reply:
Yes, I do but I haven't merged this. You can refer the following snippet:
```python
from torchtext import data

class TrecDataset(data.TabularDataset):
    dirname = 'data'
    @classmethod

    def splits(cls, question_id, question_field, answer_field, 
                    external_field, label_field, root='.data',
                    train='trecqa.train.tsv', validation='trecqa.dev.tsv',
                    test='trecqa.test.tsv'):
        path = './data'
        return super(TrecDataset, cls).splits(
            path, root, train, validation, test,
            format='TSV', fields=[('qid', question_id), ('label', label_field),
                                  ('question', question_field), ('ext_feat', external_field),
                                  ('answer', answer_field)])
```
notice the mandatory argument `root`